packages/react/src/time-picker/TimePicker.tsx

```diff
@@ -0,0 +1,101 @@
+import { useEffect, useState } from 'react';
+import { useCombobox } from 'downshift';
+import { type ComboboxProps } from '../combobox/Combobox';
+import { ComboboxBase } from '../combobox/ComboboxBase/ComboboxBase';
+import {
+	type DefaultComboboxOption,
+	useComboboxInputId,
+} from '../combobox/utils';
+import { type TimeFormat } from '../time-input/utils';
+import { filterOptions, generateOptions } from './utils';
+
+export function TimePicker({
+	id,
+	inputRef: inputRefProp,
+	interval = 15,
+	loading = false,
+	max = '24:00',
+	maxWidth = 'md',
+	min = '00:00',
+	onChange,
+	timeFormat = 'h:mm aaa',
+	value,
+	...props
+}: TimePickerProps) {
+	const {
+		// We remove this prop so that it can't be passed to ComboboxBase
+		// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+		// @ts-ignore
+		renderItem,
+		...consumerProps
+	} = props;
+	const inputId = useComboboxInputId(id);
+	const [options, setOptions] = useState<DefaultComboboxOption[]>(
+		generateOptions({ interval, max, min, timeFormat })
+	);
+	const [inputItems, setInputItems] =
+		useState<DefaultComboboxOption[]>(options);
```

Comment:
Similar to the above comment, you should be able to safely remove `<DefaultComboboxOption[]>`.